### ISSUE TITLE
Sudo: no tty present

### ISSUE THREAD

**Issue by bluegizmo83** [BusesCanFly/rpi-hunter#3]
When I try to use any payload that requires sudo, I get this error every time:

sending payload to 192.168.1.xxx
sudo: no tty present and no askpass program soecified

**Comment by BusesCanFly** [comment]
Interesting. This makes sense because you're trying to run commands as `root` instead of `pi`. What is the payload you are trying to send? Did you try wrapping it in quotes?

**Comment by bluegizmo83** [comment]
> Interesting. This makes sense because you're trying to run commands as `root` instead of `pi`. What is the payload you are trying to send? Did you try wrapping it in quotes?

I've tried one of the built in payloads `'shadow':'sudo cat /etc/shadow'` and a simple test payload that I added `'test':'sudo echo "TEST TEST TEST" \> log1.txt'`, both of which failed to execute with that sudo: no tty present and no askpass program specified error. The test payload I added works fine without using sudo, as its not needed for that command, I just added it to test if it also gave that error.

The only way I've been able to get sudo commands working thus far is to add `-S` to the command, i.e. `'test':'sudo -S echo "TEST TEST TEST" \> log1.txt'`, which will prompt me in the rpi-hunter script during the payload execution to manually enter the password, which allows the payload to execute successfully. 

Not sure if it matters, but I'm not testing against a normal stock Raspbian build... My only Raspberry Pi (a 3B) is running Octoprint, but it is based on Raspbian.

**Comment by BusesCanFly** [comment]
If I am entirely honest I haven't looked at this project in quite a while, but if you're willing to test some things I'd be happy to troubleshoot!

*Payloads:*
* Can you try specifying a payload in the cli like `--payload "sudo whoami"`? (Keeping the quotes)
* Can you try editing your test payload to be `'test':'"sudo echo "TEST TEST TEST" \> log1.txt"'`

*sudo -S:*

Can you modify line 96 of the main code to be
`os.system("sshpass -p \""+args.creds+"\" ssh -S -o stricthostkeychecking=no "+args.uname+"@"+lines[i]+" "+payload)` and see what happens? My understanding of how sshpass works isn't complete but that might work.

*OS:*

This most likely doesn't matter but I do appreciate the extra info!

**Comment by bluegizmo83** [comment]
> If I am entirely honest I haven't looked at this project in quite a while, but if you're willing to test some things I'd be happy to troubleshoot!
> 
> _Payloads:_
> 
>     * Can you try specifying a payload in the cli like `--payload "sudo whoami"`? (Keeping the quotes)
> 
>     * Can you try editing your test payload to be `'test':'"sudo echo "TEST TEST TEST" \> log1.txt"'`
> 
> 
> _sudo -S:_
> 
> Can you modify line 96 of the main code to be
> `os.system("sshpass -p \""+args.creds+"\" ssh -S -o stricthostkeychecking=no "+args.uname+"@"+lines[i]+" "+payload)` and see what happens? My understanding of how sshpass works isn't complete but that might work.
> 
> _OS:_
> 
> This most likely doesn't matter but I do appreciate the extra info!

Yeah I don't mind, and thanks for the help :)

Using the `--payload "sudo whoami"` cli command throws the same sudo tty error.

Editing my test payload to `'test':'"sudo echo "TEST TEST TEST" \> log1.txt"'` throws the same sudo tty error.

Modifying the code to `os.system("sshpass -p \""+args.creds+"\" ssh -S -o stricthostkeychecking=no "+args.uname+"@"+lines[i]+" "+payload)` does NOT throw the same sudo tty error, but now throws a different error: `ssh: Could not resolve hostname stricthostkeychecking=no: Name or service not known`

I spent a few hours trying out different things and I found a fix that doesn't require me to manually enter the password to execute sudo commands. There is probably a more elegant way of doing this, but this is what I came up with:

1. I added a new variable just below all the parser.add_argument lines. The new variable is now on line 40 and is `sudo_cmd='echo '+args.creds+' \| sudo -S '`
2. I have edited all the payloads with sudo in them to replace sudo with the new variable. For example `'test':'sudo echo "TEST TEST TEST" \> log1.txt'` now becomes `'test':sudo_cmd + 'echo "TEST TEST TEST" \> log1.txt'`

This works by still prompting for the password by using -S but it now automatically enters the password (from args.creds) without any user input.
